### PR TITLE
Add generic EntityUid, remove some usages of .Owner

### PR DESCRIPTION
**Pull request by DrSmugleaf**
Requires https://github.com/space-wizards/space-station-14/pull/21100

For the xeno antag pr btw

### CODE REVIEW

File: Robust.Shared/GameObjects/Entity.cs [hunk 1]
```diff
@@ -0,0 +1,217 @@
+﻿using Robust.Shared.Utility;
+
+namespace Robust.Shared.GameObjects;
+
+public record struct Entity<T>
```
Comment (moonheart08) [inline Robust.Shared/GameObjects/Entity.cs:5]: IEntity real?

**Review by moonheart08 — APPROVED** [review]
Approving the API changes, didn't read much else.

File: Robust.Shared/GameObjects/Entity.cs (hunk 1 above)
Comment (DrSmugleaf) [reply Robust.Shared/GameObjects/Entity.cs:5]: Return of the King

**Comment by deltanedas** [comment]
a funny i found is that you cant pass `Entity<SomethingComponent>` to a function expecting `Entity<SomethingComponent?>` you need to (probably) use `(ent.Owner, ent.Comp)` which is annoying

**Comment by deltanedas** [comment]
@DrSmugleaf is there a better way to do above

**Pull request merged**